### ISSUE TITLE
Issue attempting to display youtube videos

### ISSUE THREAD

**Issue by sstone72389** [ga-wdi-boston/capstone-project#680]
So I am trying to display youtube videos along with posts that my users make. 

I added a form as such to the handlebars:

```
        <form id="vid-post">
          <fieldset>
            <div class="form-group">
              <input class=" form-control zoom-control" placeholder="enter YouTube URL here" type="text" name="post[name]">
            </div>
            <div class="text-center">
              <button class="btn btn-transparent btn-lg" type="submit"> Upload Video </button>
            </div>
          </fieldset>
        </form>
```

When a user submits the form, it attempts to update an iframe in my HTML with the url that was submitted. I am able to get this far. When I use a YouTube URL I get this error:

```
localhost/:1 Refused to display 'https://youtu.be/fjppyqkKJI8origin=localhost:7165' in a frame because it set 'X-Frame-Options' to 'sameorigin'.
```

I tried setting the  origin as such: 

https://stackoverflow.com/questions/31442797/embedding-youtube-video-http-https-matching-error

I am able to get the videos to attach to the div by instead using an entire embed code (although this is not preferable). I used the following code:

```
$('.vid-content').append(response.post.name)
```

where response.post.name is the submitted embed code...

I have a few issues here that are troubling me, but the main one being that the videos do not remain visible if logged out, and back in. Maybe I need some kind of ajax request to get them and THEN append them to the div? I have tried a couple times and cannot get it to work.

I could really use some help with this. I have been battling this for a couple days.

**Comment by MicFin** [comment]
That URL doesn't look correct.

As an example URL we have this fantastic cat video 
`https://www.youtube.com/watch?v=hY7m5jjJ9mM`

Ideally YouTube videos want to be embedded in HTML as:
```html
<iframe width="560" height="315" src="https://www.youtube.com/embed/hY7m5jjJ9mM" frameborder="0" allowfullscreen></iframe>
```

Notice that the URL that YouTube wants to embed is not the same as the URL a user goes to but it is a specific URL for embedding in an iframe.
`https://www.youtube.com/embed/hY7m5jjJ9mM`

However, the last part of the URL is the same `hY7m5jjJ9mM`.

Potentially the user can submit the whole URL `https://www.youtube.com/watch?v=hY7m5jjJ9mM` but on the back end you only save the last part `hY7m5jjJ9mM`.  This might be helpful for selecting just the last part of the URL https://stackoverflow.com/questions/6903823/regex-for-youtube-id.

Then you can use handlebars to add the ending of the URL to your iFrame.
```html
<iframe width="560" height="315" src="https://www.youtube.com/embed/{{ post.video_url}}" frameborder="0" allowfullscreen></iframe>
```

**Comment by tvcrawley** [comment]
@sstone72389 Update?

**Comment by sstone72389** [comment]
Okay, so I have a few ideas on how to implement this, per Mikes post. I am going to pay around with this now and provide a technical update, shortly.

**Comment by sstone72389** [comment]
I was able to resolve this issue by removing the form from handlebars, and requiring an embed link in the initial creation of a post (user input). In handlebars I was able to escape the string and display the video by using `{{{post.name}}}` , where post.name is the HTML input from the user (yes, I know the naming convention is weird haha. that is next to fix). The triple curly braces allows it to display as HTML instead of a string. Using embed links is standard for such practices, so I decided this was both the easiest method to use and adheres to the expected process for the user.
